apps/web/src/pages/settings/organization/OrganizationLogo.tsx

```diff
@@ -0,0 +1,90 @@
+import { errorMessage, successMessage } from '@novu/design-system';
+import { IResponseError } from '@novu/shared';
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+import React, { useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useUpdateOrganizationBranding } from '../../../api/hooks';
+import { getSignedUrl } from '../../../api/storage';
+import { ProfileImage } from '../../../components/shared';
+
+type FormValues = {
+  logo: string;
+};
+
+const MIME_TYPES = {
+  'image/jpeg': 'jpeg',
+  'image/png': 'png',
+};
+
+export function OrganizationLogo({ logo }: { logo?: string }) {
+  const { updateOrganizationBranding } = useUpdateOrganizationBranding();
+  const { control, reset } = useForm<FormValues>({
+    defaultValues: {
+      logo,
+    },
+  });
+
+  /**
+   * Reset the form when the organization changes
+   * Useful in the scenario when the user switches to a different organization
+   */
+  useEffect(() => {
+    reset({ logo });
+  }, [logo, reset]);
+
+  const { mutateAsync: getSignedUrlAction } = useMutation<
+    { signedUrl: string; path: string; additionalHeaders: object },
+    IResponseError,
+    string
+  >(getSignedUrl);
+
+  async function handleUpload(event: React.ChangeEvent<HTMLInputElement>) {
+    const file = event.target.files?.[0] ? event.target.files[0] : null;
+
+    if (!file) {
+      return;
+    }
+
+    const fileExtension = file.type;
+    const isValidFileType = MIME_TYPES[fileExtension] !== undefined;
+    if (!isValidFileType) {
+      errorMessage('Invalid file type. Please upload a PNG or JPEG file.');
+
+      return;
+    }
+
+    const { signedUrl, path, additionalHeaders } = await getSignedUrlAction(MIME_TYPES[fileExtension]);
+    const contentTypeHeaders = {
+      'Content-Type': fileExtension,
+    };
+
+    const mergedHeaders = Object.assign({}, contentTypeHeaders, additionalHeaders || {});
+    await axios.put(signedUrl, file, {
+      headers: mergedHeaders,
+      transformRequest: [
+        (data, headers) => {
+          if (headers) {
+            delete headers.Authorization;
+          }
+
+          return data;
+        },
+      ],
+    });
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** This chunk seems like it could be an independent async function defined separately from the component. This improves testability, separation of responsibilities, and makes the component smaller